src/middlewares/ratingValidation.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-underscore-dangle */
+import Joi from 'joi';
+import jwt from 'jsonwebtoken';
+import { config } from 'dotenv';
+
+config();
+
+const validateRatingData = (req, res, next) => {
+  const schema = Joi.object().keys({
+    rating: Joi.number().required(),
+  });
+
+  const error = (['POST'].includes(req.method))
+    ? Joi.validate(req.body, schema).error
+    : null;
+  if (error) {
+    return res.status(400).json({ message: error.details[0].message });
+  }
+  next();
+};
+
+const validateUser = (req, res, next) => {
+  const loggedInUser = req.headers.authorization;
+  const validUser = loggedInUser ? jwt.decode(loggedInUser, process.env.JWT_KEY, { expiresIn: '10h' }) : false;
+  const { id } = validUser;
+  req.userId = id;
+  if (validUser) {
+    return next();
+  }
```

Comment:
I am not sure, but I think one of the authentication middlewares adds a user property to the request, please check the authentication middleware to see if it does or not. Then you would just need to do a `const { id } = req.user` instead of having to decode the auth token again